### PR TITLE
[receiver/kubeletstats] Update scope name to otelcol/kubeletstatsreceiver

### LINKED ISSUE #9399: update receiver/kubeletstats to use NewMetricData

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Update Scope Name to otelcol/kubeletstatsreceiver in all scenarios.  Update to v2 metrics builder will happen in a different PR.

**Link to tracking Issue:**
Fixes #9399 

**Testing:**
Updated unit tests to check for scope name.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - `stackdriverexporter`: Remove the stackdriver exporter in favor of the identical googlecloud exporter (#9274)
 - `filelog`, `journald`, `syslog`, `tcplog`, `udplog`: Remove `preserve_to` field from sub-parsers (#9331)
+- `kubeletstatsreceiver`: instrumentation name updated from `otelcol/kubeletstats` to `otelcol/kubeletstatsreceiver` (#9400)
```
Comment (dmitryax) [inline CHANGELOG.md:9]: I believe `kubeletstats` currently. Just to be precise
```suggestion
- `kubeletstatsreceiver`: instrumentation name updated from `kubeletstats` to `otelcol/kubeletstatsreceiver` (#9400)
```

Comment (TylerHelmuth) [reply CHANGELOG.md]: You're right

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Looks like a related test is failing

**Comment by TylerHelmuth** [comment]
> Looks like a related test is failing

Ya that k8sclient test is flaky.  Being tracked in #9091.

I think the  check-collector-module-version test is failing on every branch right now.

**Comment by codeboten** [comment]
this should fix the check-collector-module-version: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9402/files

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**